### PR TITLE
[DOCS] Adds descriptions to ML APIs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

Related to https://github.com/elastic/elasticsearch-specification/issues/964.

Adds descriptions to ML API properties. Source of info: 
* https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-apis.html
* https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-ad-apis.html
* https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-df-analytics-apis.html
* https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-df-trained-models-apis.html

### CODE REVIEW

File: specification/ml/_types/Job.ts [hunk 1]
```diff
@@ -59,63 +59,273 @@ export class JobStatistics {
...
   model_snapshot_retention_days: long
+  /**
+   * Advanced configuration option.
+   * The period over which adjustments to the score are applied, as new data is seen.
+   * The default value is the longer of 30 days or 100 `bucket_spans`.
+   */
   renormalization_window_days?: long
+  /**
+   * A text string that affects the name of the machine learning results index.
+   * The default value is `shared`, which generates an index named `.ml-anomalies-shared`.
```
Comment (abdonpijpelink) [inline specification/ml/_types/Job.ts:167]:
```suggestion
   * The default value is `shared`, which generates an index named `.ml-anomalies-shared`.
   * @server_default shared
```

File: specification/ml/evaluate_data_frame/types.ts [hunk 2]
```diff
@@ -22,24 +22,65 @@ import { Name } from '@_types/common'
 import { double, integer } from '@_types/Numeric'
 
 export class DataframeOutlierDetectionSummary {
+  /**
+   * The AUC ROC (area under the curve of the receiver operating characteristic) score and optionally the curve.
+   * @server_default "include_curve": false
```
Comment (abdonpijpelink) [inline specification/ml/evaluate_data_frame/types.ts:27]: I notice that the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/master/evaluate-dfanalytics.html#oldetection-resources) surround the default value by curly brackets. Should this perhaps be `{"include_curve": false}`?

**Review by abdonpijpelink — APPROVED** [review]
LGTM. I've left a few small suggestions.

File: specification/ml/_types/Job.ts (hunk 1 above)
Comment (szabosteve) [reply specification/ml/_types/Job.ts:167]: I intentionally left the default values out here as the generator complains if I add a default value for an optional parameter.

File: specification/ml/_types/Job.ts [hunk 3]
```diff
@@ -59,63 +59,273 @@ export class JobStatistics {
...
+   * It provides a more detailed view into anomaly detection.
+   * Model plot provides a simplified and indicative view of the model and its bounds.
+   */
   model_plot_config?: ModelPlotConfig
   model_snapshot_id?: Id
+  /**
+   * Advanced configuration option, which affects the automatic removal of old model snapshots for this job.
+   * It specifies the maximum period of time (in days) that snapshots are retained.
+   * This period is relative to the timestamp of the most recent snapshot for this job.
+   * The default value is `10`, which means snapshots ten days older than the newest snapshot are deleted.
```
[the remark this answers is not in the document]
Comment (szabosteve) [reply specification/ml/_types/Job.ts]: I need to remove the default value as the generator complains about it:

```
Error: Default values can only be specified on optional properties
At: /Users/szabosteve/Repos/elasticsearch-specification/specification/ml/_types/Job.ts:152:3
```
150    model_plot_config?: ModelPlotConfig
151    model_snapshot_id?: Id
152    /**
153     * Advanced configuration option, which affects the automatic removal of old model snapshots for this job.
154     * It specifies the maximum period of time (in days) that snapshots are retained.
155     * This period is relative to the timestamp of the most recent snapshot for this job.
156     * By default, snapshots ten days older than the newest snapshot are deleted.
157     * @server_default 10
158     */
159    model_snapshot_retention_days: long
160    /**
```

File: specification/ml/evaluate_data_frame/types.ts (hunk 2 above)
Comment (szabosteve) [reply specification/ml/evaluate_data_frame/types.ts]: Thank you for the review! I updated the PR via [68e51f0](https://github.com/elastic/elasticsearch-specification/pull/2245/commits/68e51f00dd8fe63db4a158cd557979d437ae6e57).

**Pull request merged**